cloud-media-livestream/keypublisher/README.md

```diff
@@ -0,0 +1,302 @@
+included here. Copy the template file and replace any needed values. In
+particular `CLOUD_FUNCTION_URL` must be replaced with the URL you received when
+you deployed your function.
+
+Once the file has been created, create your API config:
+
+```shell
+gcloud api-gateway api-configs create livestream-key-publisher-api-config-1 \
+  --display-name="Live Stream Key Publisher API Config v1" \
+  --api=livestream-key-publisher-api \
+  --openapi-spec="api-config.yml" \
+  --backend-auth-service-account="livestream-cf-invoker@[MY-PROJECT].iam.gserviceaccount.com"
+```
+
+Next, create the API Gateway:
+
+```shell
+gcloud api-gateway gateways create livestream-key-publisher-api-gateway \
+  --display-name="Live Stream Key Publisher API Gateway" \
+  --api=livestream-key-publisher-api \
+  --api-config=livestream-key-publisher-api-config-1 \
+  --location=us-central1
+```
+
+Once created, your API must be enabled. Run a command to describe your API:
+
+```shell
+gcloud api-gateway apis describe livestream-key-publisher-api
+```
+
+You will get the `managedService` field in the response. This is the service you
+need to enable. For example,
+
+```shell
+gcloud services enable livestream-key-publisher-api-144h6p0e7bzc1.apigateway.[MY-PROJECT].cloud.goog
+```
+
+Now you need to locate the endpoint URL of your API. Run a command to describe
+your gateway:
+
+```shell
+gcloud api-gateway gateways describe livestream-key-publisher-api-gateway --location=us-central1
+```
+
+You will get the `defaultHostname` field in the response. For example,
+
+```
+defaultHostname: livestream-key-publisher-api-gateway-48mwfke9.uc.gateway.dev
+```
+
+You will send API requests to this URL.
+
+Create an API key to use for making queries to your API, and restrict the API
+key such that only your newly created `livestream-key-publisher-api` can be
+called with the API key. Specify the API restriction with the `managedService`
+field from earlier:
+
+```shell
+gcloud alpha services api-keys create --api-target=service=[managedService]
+```
+
+You will get the `keyString` field in the response. This is your API key. Save
+this key, as you will not be able to retrieve it again.
+
+### Test your API
+
+Using the above information, use `curl` to make a `GET` query to your API:
+
+```shell
+curl --location --request POST \
+  'https://livestream-key-publisher-api-gateway-48mwfke9.uc.gateway.dev/keys?api_key=YOUR_API_KEY' \
+  --header 'Content-Type: application/json' \
+  --data '{"mediaId": "my-asset", "provider": "FakeProvider", "keyIds": ["abcd1234", "efgi5678"]}'
+```
+
+Arguments in the request body are:
+
+*   `mediaId`: arbitrary identifier for the media being encrypted.
+*   `provider`: the DRM provider to use. Omit this argument to see a list of
+    supported providers.
+*   `keyIds`: a list of key IDs to prepare for the given media ID.
+
+This example query uses `FakeProvider`. This is a provider used as a reference
```

Comment:
What should the user precisely need to update? Is it just `FakeProvider`? calling it out explicitly will definitely help.

Reply:
I've updated the instruction. All arguments can be updated, especially the `provider` and `keyIds` fields. The code in this sample should help users who are working with a third-party DRM provider and need a reference for integrating with Live Stream API. So they should have `keyIds` information provided by their DRM provider already.